tests/nlu/emulators/test_luis.py

```diff
@@ -19,26 +18,103 @@ def test_luis_response():
             {"confidence": 0.08816417744097163, "name": "greet"},
             {"confidence": 0.058766588386123204, "name": "affirm"},
         ],
-        "entities": [{"entity": "cuisine", "value": "italian"}],
+        "entities": [
+            {
+                "entity": "cuisine",
+                "value": "italian",
+                "role": "roleCuisine",
+                "extractor": "SpacyEntityExtractor",
+                "start": 7,
+                "end": 14,
+            }
+        ],
     }
     norm = em.normalise_response_json(data)
     assert norm == {
         "query": data["text"],
-        "topScoringIntent": {"intent": "restaurant_search", "score": 0.737014589341683},
-        "intents": [
-            {"intent": "restaurant_search", "score": 0.737014589341683},
-            {"intent": "goodbye", "score": 0.11605464483122209},
-            {"intent": "greet", "score": 0.08816417744097163},
-            {"intent": "affirm", "score": 0.058766588386123204},
-        ],
-        "entities": [
-            {
-                "entity": e["value"],
-                "type": e["entity"],
-                "startIndex": None,
-                "endIndex": None,
-                "score": None,
-            }
-            for e in data["entities"]
+        "prediction": {
+            "normalizedQuery": data["text"],
+            "topIntent": "restaurant_search",
+            "intents": {
+                "restaurant_search": {"score": 0.737014589341683},
+                "goodbye": {"score": 0.11605464483122209},
+                "greet": {"score": 0.08816417744097163},
+                "affirm": {"score": 0.058766588386123204},
+            },
+            "entities": {
+                "roleCuisine": ["italian"],
+                "$instance": {
+                    "role": "roleCuisine",
+                    "type": "cuisine",
+                    "text": "italian",
+                    "startIndex": 7,
+                    "length": len("italian"),
+                    "score": None,
+                    "modelType": "SpacyEntityExtractor",
+                },
+            },
+        },
+    }
+
+
+def test_luis_response_without_role():
+    em = LUISEmulator()
+    data = {
+        "text": "I want italian food",
+        "intent": {"name": "restaurant_search", "confidence": 0.737014589341683},
+        "intent_ranking": [
+            {"confidence": 0.737014589341683, "name": "restaurant_search"}
         ],
+        "entities": [{"entity": "cuisine", "value": "italian"}],
+    }
+    norm = em.normalise_response_json(data)
+    assert norm == {
+        "query": data["text"],
+        "prediction": {
+            "normalizedQuery": data["text"],
+            "topIntent": "restaurant_search",
+            "intents": {"restaurant_search": {"score": 0.737014589341683}},
+            "entities": {
```

Comment:
The entities still don't match what they describe in the documentation 🤔 

```
"entities":{
    "Destination":[
        "Yellow Bird Lane"
    ],
    "$instance":{
        "Destination": [
            {
                "role": "Destination",
                "type": "Location",
                "text": "Yellow Bird Lane",
                "startIndex": 25,
                "length":16,
                "score": 0.9837309,
                "modelTypeId": 1,
                "modelType": "Entity Extractor"
            }
        ]
    }
}
```

Reply:
Yes, you're right. Updated it